aws/components/directory/setup.ftl

```diff
@@ -0,0 +1,221 @@
+                        "SecretKey" : passwordSecretKey
+                    }
+                }
+            componentType=DIRECTORY_COMPONENT_TYPE
+        /]
+    [#else]
+        [@fatal
+            message="Could not find link to secret store or link was invalid"
+            detail="Add a link to a secret store component which will manage the root credentials"
+            context=solution.RootCredentials.SecretStore
+        /]
+    [/#if]
+
+    [#-- Output Generation --]
+    [#if deploymentSubsetRequired(DIRECTORY_COMPONENT_TYPE, true)]
+
+        [#-- Network Security --]
+        [@createSecurityGroup
+            id=securityGroupId
+            name=securityGroupName
+            vpcId=vpcId
+            occurrence=occurrence
+        /]
+
+        [@createSecurityGroupRulesFromNetworkProfile
+            occurrence=occurrence
+            groupId=securityGroupId
+            networkProfile=networkProfile
+            inboundPorts=dnsPorts
+        /]
+
+        [#local ingressNetworkRule = {
+                "Ports" : dnsPorts,
+                "IPAddressGroups" : solution.IPAddressGroups
+        }]
+
+        [@createSecurityGroupIngressFromNetworkRule
+            occurrence=occurrence
+            groupId=securityGroupId
+            networkRule=ingressNetworkRule
+        /]
+
+        [#if !hibernate]
+            [#local vpcSettings= {
+                    "SubnetIds" : getSubnets(core.Tier, networkResources),
+                    "VpcId" : getReference(vpcId)
+                }]
+
+            [#-- Component Specific Resources --]
+            [@createDSInstance
+                id=dsId
+                name=dsShortName
+                type=type
+                masterPassword=getSecretManagerSecretRef(resources["rootCredentials"]["secret"].Id, "password")
+                enableSSO=enableSSO
+                fqdName=fqdName
+                shortName=shortName
+                size=size
+                vpcSettings=vpcSettings
+            /]
+
+            [@cfOutput
+                formatId(dsId, IP_ADDRESS_ATTRIBUTE_TYPE), 
+                {
+                    "Fn::Join": [
+                        ",",
+                        {
+                            "Fn::GetAtt": [
+                                dsId,
+                                "DnsIpAddresses"
+                            ]
+                        }
+                    ]
+                },
+                false
+            /]
+
+[#--
+            [@cfOutput
+                formatId(dsId, ALIAS_ATTRIBUTE_TYPE), 
+                {
+                    "Fn::Join": [
+                        ",",
+                        {
+                            "Fn::GetAtt": [
+                                dsId,
+                                "Alias"
+                            ]
+                        }
+                    ]
+                },
+                false
+            /]
+--]
```

Comment:
Do we need this? What would the IP addresses be used for? The dhcp scopes? and we can probably remove the commented out one or add comments why its like that

Reply:
The IP addresses of the DNS servers for the directory, I was thinking about the possibility of pulling these on a windows server being created with a link to the DS

The commented out block has to do with the note in the description about my deliberately not included the createAlias feature. Left it there in case you wanted to say it was needed

Reply:
Capturing the output won't change if its defined or not, just that it would be saved in our local outputs. Are you saying you haven't implemented the alis configuration?

Reply:
OK, so we could still access it from the console.

Correct, I haven't implemented the alias configuration. Should I?